commands/cli.go

```diff
@@ -257,6 +257,75 @@ func BuildContextualMenu() []cli.Command {
 				return nil
 			},
 		},
+		{
+			Name:     "read",
+			Usage:    "Decrypt and display a named secret",
+			Category: statecategory,
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:   "key",
+					EnvVar: "SCTL_KEY",
+					Usage:  "KMS Key URI",
+				},
+			},
+			Action: func(c *cli.Context) error {
+				ctxerr := validateContext(c, "read")
+				if ctxerr != nil {
+					return ctxerr
+				}
+
+				secrets, keyURI, err := utils.ReadSecrets()
+				if err != nil {
+					return err
+				}
+
+				searchTerm := c.Args().First()
+
+				toDecrypt, findErr := secrets.Find(strings.ToUpper(searchTerm))
+				if findErr != nil {
+					return findErr
+				}
+
+				locatedSecret := secrets[toDecrypt]
+
+				// uncan the base64
+				decoded, err := base64.StdEncoding.DecodeString(locatedSecret.Cyphertext)
+				if err != nil {
+					return errors.Wrap(err, "failed secret decode")
+				}
+				// Work with the envelope's provided key or switch to CLI flags/env
+				var client cloud.KMS
+				if keyURI == "" {
+					log.Warn("No KeyURI found in envelope. Required usage of flag/env config.")
+					err := validateContext(c, "read")
```

Comment:
`validateContext` is called both here and at the beginning of the read handler fn (L272) -- is this desired?

Reply:
So this gave me an idea for a better pattern that i used in the read function post review-feedback. I swap the context evaluation between default and named, which deviates just a slight bit from how it was being used before.